lib/fieldhand/repository.rb

```diff
@@ -14,22 +14,26 @@ module Fieldhand
   #
   # See https://www.openarchives.org/OAI/openarchivesprotocol.html
   class Repository
-    attr_reader :uri, :logger, :timeout, :headers
+    attr_reader :uri, :logger, :timeout, :retries, :interval, :headers
```

Comment:
Rather than having to repeat each option in both the `Repository` and the `Paginator`, I wonder if we could simplify things a little by building an `Options` object here and then passing _that_ directly to the `Paginator` so we avoid having to parse out the `retries` and `interval` twice? That would mean that `retries` is no longer part of the `Repository` API though.

The tricky bit would be to make this compatible with the existing `Repository` and `Paginator` constructor (if we assume the `Paginator` constructor is part of the gem's public API).

Reply:
If we are happy to remove the single calls from the Repository API, all could be delegated to options (and we would avoid readers repetition). i.e:

```ruby
class Repository
    attr_reader :uri, :options
    #[...]

def initialize(uri, logger_or_options = {})
    @uri = uri.is_a?(::URI) ? uri : URI(uri)

    @options = Options.new(logger_or_options)
end
```

The constructor could be become:
```ruby
def paginator
    @paginator ||= Paginator.new(
      uri,
      :logger => options.logger,
      :timeout => options.timeout,
      # [...]
end
```

I could not find a better solution for the `Paginator` constructor signature, considering we might pass a `Hash`, an `Options` object or a `Logger`. `Options` already encapsulates logic to parse the arguments it doesn't make a lot of sense to add more (correct me if I am wrong)

Reply:
Given that `logger`, `timeout` and `headers` are only taken by the `Repository` to pass to the `Paginator`, could we perhaps pass the whole of `logger_or_options` in `Repository#initialize` to `Paginator#initialize` and delegate `logger`, `timeout` and `headers` from `Repository` to `Repository#paginator` for backward compatibility, e.g.

```ruby
require 'forwardable'

class Repository
  attr_reader :uri, :logger_or_options

  extend Forwardable
  def_delegators :paginator, :logger, :timeout, :headers

  def initialize(uri, logger_or_options = {})
    # ...

    @logger_or_options = logger_or_options
  end

  # ...

  private

  def paginator
    @paginator ||= Paginator.new(uri, logger_or_options)
  end
end
```